src/consensus/dao.cpp

```diff
@@ -2645,7 +2645,7 @@ bool CPaymentRequest::IsAccepted(const CStateViewCache& view) const
     if (nVersion & ABSTAIN_VOTE_VERSION)
         nTotalVotes += nVotesAbs;
 
-    return nTotalVotes > (GetConsensusParameter(Consensus::CONSENSUS_PARAM_VOTING_CYCLE_LENGTH, view) * nMinimumQuorum) - exclude
+    return nTotalVotes > ((GetConsensusParameter(Consensus::CONSENSUS_PARAM_VOTING_CYCLE_LENGTH, view) - exclude) * nMinimumQuorum)
             && ((float)nVotesYes > ((float)(nTotalVotes) * GetConsensusParameter(Consensus::CONSENSUS_PARAM_PAYMENT_REQUEST_MIN_ACCEPT, view) / 10000.0));
```

Comment:
The second check currently could be reverted if no votes are increasing all of a sudden. Shouldn't the second check be something like this? 
`nVoteYes > (nTotalVotes + remaining blocks in the cycle) * GetConsensusParameter(Consensus::CONSENSUS_PARAM_PAYMENT_REQUEST_MIN_ACCEPT, view) / 10000.0)`

Reply:
id say because of the first check (nTotalVotes is larger than the min quorum), nTotalVotes will always have a value in a valid range so i don't see the need to add the remaining blocks in the cycle. but maybe i'm understanding it wrong?

Reply:
The current checks are correct logically. The only thing that came to my mind is that there will be cases where the checks pass and the text are displayed like "accepted, waiting for end of voting cycle". It is not technically correct since there is still a chance where the no votes suddenly increase and overthrow the decision. 
For example,
If block cycle has 10 blocks, first 5 are yeses, then the checks will pass and the text will be displayed. However, there is still a chance where the remaining 5 could overthrow that outcome. In this scenario, the proposed checks will not pass so it'd still be pending.

All in all, I'd say the proposed check may not be needed if we just change the displayed text to show something different than "Accepted....." since it's not accepted yet.

Reply:
mmm i would say the message is only shown when there are 50%+1 votes (because of the > sign) and therefor it cant be reversed

Reply:
it can since the required yes votes are 70%. So cycle of 10 blocks, first 6 blocks are yes, at this stage it passes both checks (pass 50% quorum and 70% yes) so it'd show accepted. But if the remaining 4 blocks are all no votes, then we would end up having only 60% approval and the accepted is reverted right?

Reply:
ok now i got it. you are right

Reply:
so you mean something more like this:
```c++
(float)nVotesYes > (float)(GetConsensusParameter(Consensus::CONSENSUS_PARAM_VOTING_CYCLE_LENGTH, view) - exclude) * GetConsensusParameter(Consensus::CONSENSUS_PARAM_PAYMENT_REQUEST_MIN_ACCEPT, view) / 10000.0
```

Reply:
not exactly, this will result in a static amount of yes votes needed while the proposed one is more dynamic basing of current voting conditions. 
For example, with this check, assuming there are no exclude votes, required yes votes are always 7 in the case of 10 block cycle. The proposed one could have much less than 7 votes. 
Say the first 5 votes are no votes (not abstain and not exclude but just not voting), and adding 4 yes votes, the proposed check will pass while this check won't pass.

Reply:
the problem is that this function can't see what is the current block inside of the cycle so it can't know how many blocks are left

Reply:
i see. maybe we can change the text then? Like to "en-route to acceptance" for something in those lines?

Reply:
what about just "pending" when the cycle did not end

Reply:
yes that works as well. is the text using the same checks here? How does it work when these checks changed from false to true but it's pending all the way?

Reply:
it would check for a state change which would only happen after the end of a cycle

Reply:
sounds good.